File: 287-find-the-duplicate-number/find-the-duplicate-number.py
class Solution(object):
    def findDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        n=len(nums)
        ans=0
        for i in range(31):
            x=0
            y=0
            for j in range(n):
                if nums[j] & (1<<i)!=0:
                    x+=1
                if j>=1 and (j&(1<<i))!=0:
                    y+=1
            if x>y:
                ans|=(1<<i)
        return ans
       
        
        

File: 287-find-the-duplicate-number/find-the-duplicate-number.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        
        int n=nums.size();
        vector<bool> vis(n+1,false);
        for(int i=0;i<n;i++){
            if(vis[nums[i]]) return nums[i];
            vis[nums[i]]=true;
        } 
        return -1;
        
    }
};